server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -414,6 +443,9 @@ private void handleApplyCommit(ApplyCommitRequest applyCommitRequest, ActionList
             logger.trace("handleApplyCommit: applying commit {}", applyCommitRequest);
 
             coordinationState.get().handleCommit(applyCommitRequest);
+            // Since the coordination state changed, we update accordingly
```

Comment:
These comments seem unnecessary (and indeed positively distracting) to me - really it's not saying much more than "the next line calls `updateClusterFormationClusterStateView()` because it has to" but such comments could be applied to basically every line of code here.